### DISCUSSION TITLE
Confused: agent_nodepools and autoscaler_nodepools

### DISCUSSION THREAD

**Discussion by tgrushka** [kube-hetzner/terraform-hcloud-kube-hetzner#1235] [Q&A]
Thanks for this project!

Not clear from docs: Are `agent_nodepools` and `autoscaler_nodepools` complementary or distinct? Are they both worker nodepools, where one has CA autoscaling enabled and the other does not? Can I leave `agent_nodepools` empty and put all my nodepools in `autoscaler_nodepools` if I want to be able to scale all of them, and just set `min_nodes = 0` and `max_nodes = 0` to not use for now, and `min_nodes = 1` and `max_nodes = 1` to not autoscale a particular pool?

Or, do I need to define all the nodepools in `agent_nodepools` and then duplicate them in `autoscaler_nodepools`?

Please also clarify this in the README and example.tf.

Also, maybe note that **agent nodes** == **worker nodes** as someone at some point decided to muddy the waters by introducing both terms.

Another related question:
Why not just have `agent_nodepools` combined, and set autoscaling to true or false on each one, with min/max / count ... or better yet just forget count and have min/max and CA enabled on all by default (maybe this is a resource question)?

**Comment by pat-s** [comment]
I was also wondering about this.
While `agent_nodepools` cannot be undefined, one can set `count=0` in the first node pool definition and by that effectively avoid deploying non-autoscaled nodepools.

I don't see any downside in only deploying autoscaled pools with a sensible `min_size`.

This point deserves some/more discussion in the current template.
